src/app/components/GoogleSheetsInput.svelte

```diff
@@ -0,0 +1,179 @@
+
+  /**
+   *  On load, called to load the auth2 library and API client library.
+   */
+  function handleClientLoad() {
+    gapi.load("client:auth2", initClient);
+  }
+
+  /**
+   *  Initializes the API client library and sets up sign-in state
+   *  listeners.
+   */
+  function initClient() {
+    error = null;
+    gapi.client
+      .init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES,
+      })
+      .then(
+        function () {
+          // Listen for sign-in state changes.
+          gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+          // Handle the initial sign-in state.
+          updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        },
+        function (error) {
+          error = JSON.stringify(error, null, 2);
+        }
+      );
+  }
+
+  /**
+   *  Called when the signed in status changes, to update the UI
+   *  appropriately. After a sign-in, the API is called.
+   */
+  function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+      getValuesFromSpreadSheet();
+    } else {
+      handleAuthClick();
+      getValuesFromSpreadSheet();
+    }
+  }
+
+  /**
+   *  Sign in the user upon button click.
+   */
+  function handleAuthClick(event) {
+    gapi.auth2.getAuthInstance().signIn();
+  }
+
+  /**
+   *  Sign out the user upon button click.
+   */
+  function handleSignoutClick(event) {
+    gapi.auth2.getAuthInstance().signOut();
+  }
+
+  /**
+   * Gets the unique spreadsheet ID from a Google Sheet URL
+   *
+   */
+  function getSpreadsheetId() {
+    let url = document.getElementById("googleSheetsURL").value;
+    if (url) {
+      return url.split("/")[5];
+    } else {
+      error =
+        "Error: The url " +
+        url +
+        " is not a valid Google Sheets URl, please paste a valid one.";
+    }
+  }
+
+  /**
+   * Read word and count form storage
+   * Sample: https://docs.google.com/spreadsheets/d/1lzHxoMWHpdGecby4d0y15AVf80csA7iNkaMMEL54Q7g/edit#gid=0
+   */
+  async function getValuesFromSpreadSheet() {
+    const wordListObject = [];
+    spreadsheetId = getSpreadsheetId() || "";
+    if (!error) {
+      gapi.client.sheets.spreadsheets.values
+        .get({
+          spreadsheetId: spreadsheetId,
+          range: "A2:B",
+        })
+        .then(
+          async function (response) {
+            var range = response.result;
+            if (range.values.length > 0) {
+              for (let i = 0; i < range.values.length; i++) {
+                let row = range.values[i];
+                let word = row[0];
+                let wordCount = row[1];
+                if (!wordCount) {
+                  wordCount = 0;
+                }
+                wordListObject.push([word, wordCount]);
+              }
+              error = null;
+              console.log(wordListObject); // TODO: remove me
+
+              const kmpFile = await worker.readGoogleSheet(
```

Comment:
this local constant `kmpFile` seems to be used here, if this line was intended to call readGoogleSheet function, const kmpFile can be removed 
```suggestion
              await worker.readGoogleSheet(
```

Reply:
A disturbing thing here is that this should fail type-checking. The `PredIctiveTextWorker` interface does not have the method `readGoogleSheet()` (the concrete implementation has it though). ¯\\\_(ツ)\_/¯